mailbox/jpa/src/main/java/org/apache/james/mailbox/jpa/mail/JPAMailboxMapper.java

```diff
@@ -193,13 +201,15 @@ private JPAMailbox loadJpaMailbox(MailboxId id) throws MailboxNotFoundException
                 .setParameter("userParam", mailbox.getUser().asString())
                 .getSingleResult()))
             .filter(numberOfChildMailboxes -> numberOfChildMailboxes > 0)
-            .hasElement();
+            .hasElement()
+            .subscribeOn(Schedulers.elastic());
```

Comment:
That is evil to me to enforce the scheduler in the implementation. It leads to resoure waste and I noticed clear performance gains removing all these nasty schedule on.